### ISSUE TITLE
Maybe you want to at least mention elephas in your readme

### ISSUE THREAD

**Issue by maxpumperla** [JoeriHermans/dist-keras#4]
Hi @JoeriHermans,

I like a lot what you did with this library. However, I have to say that it's fairly obvious from your commit log that you started from elephas and slowly refactored the project to make it something else, see e.g.

https://github.com/JoeriHermans/dist-keras/commit/8578a7b3f5c0f9440e7bec5a86a364ddc3baf9d9

That is in itself really not a problem, given elephas' license, but it'd only be fair to mention that you did not exactly start from scratch here. Given that you used an Elephas github issue to advertise your project I think you should follow the unwritten rule of OSS and mark a fork as a fork, even if it now evolved into an independent project.

It would have really been great to just work together on one project (e.g. by making you collaborator of elephas), instead of duplicating efforts...

Alright, but as I said in the beginning you did a great job here, so keep it up.

**Comment by JoeriHermans** [comment]
Hi @maxpumperla 

I started from Elephas since this was at first a specific request. So I first tried to fix the code in order to ensure compatibility with Keras 1.0+ and then do a pull request. But since I just started playing around with Keras internals I obviously failed to do that. Later, I started completely from scratch as can be observed in this commit: https://github.com/JoeriHermans/dist-keras/commit/92b4f63d6a49d590a6ac6d05564e786d8c00f4c3  At first I mentioned Elephas in the project heading (I can't prove that so you need to take my word for it), but after the commit mentioned above I removed it, since the codebase isn't based on Elephas at all (test.py was the experiment file at that time).

Ok, to be fair, it is kinda of an advertisement, but subtle :) However, I really have an issue with running Elephas on a cluster, local runs fine though. Any suggestions?

I know, and that was originally the plan (in the form of a pull request), but the project kinda evolved as mentioned above. Furthermore, they wanted multiple distributed optimizers, for which I needed to refactor the whole code.

I'll add you to the references if this is ok?

Kind regards,

Joeri

**Comment by maxpumperla** [comment]
Thanks Joeri, that's very nice of you. And to be clear I didn't mean to sound harsh, you obviously put a lot of work and originality into `dist-keras`. 

Honestly, the only thing that's really a pity to me is that we didn't work together from the start. Well, maybe we still can in some way. As you see from my commits for elephas I'm rather busy lately and was looking for a collaborator for elephas from the start. 


How about this? Otherwise just plainly cite the repo, I don't really mind. 

@misc{pumperla2015,
author = {Max Pumperla},
title = {elephas},
year = {2015},
publisher = {GitHub},
journal = {GitHub repository},
howpublished = {\url{https://github.com/fchollet/keras}}
}

Cheers,
Max

**Comment by JoeriHermans** [comment]
Hi Max,

I added you reference in APA format to the development branch (https://github.com/JoeriHermans/dist-keras/tree/development).

Yeah, maybe we could do something like this. I'll check if I can fix your issue regarding running an example on the cluster, I'll do a PR when I fixed it :) If I can remember correctly, I had a similar issue in the past.

**Edit**: I'll add the BibTex reference in there as well, might be useful in the future.

Kind regards,

Joeri

P.s. I'll close this issue if the reference in the development branch is ok for you.
